src/circulating_supply/user_balance.rs

```diff
@@ -0,0 +1,37 @@
+use actix::Addr;
+use near_client::{Query, ViewClientActor};
+use near_indexer::near_primitives::types::{Balance, BlockId, BlockReference};
+use near_indexer::near_primitives::views::{QueryRequest, QueryResponseKind};
+
+pub async fn get_user_balance(
+    view_client: Addr<ViewClientActor>,
+    account_id: &str,
+    block_height: u64,
+) -> Balance {
+    let block_reference = BlockReference::BlockId(BlockId::Height(block_height));
+    let request = QueryRequest::ViewAccount {
+        account_id: account_id.parse().unwrap(),
```

Comment:
I don't insist on the error message, you may change it to be more helpful

```suggestion
        account_id: account_id.parse().expect("Failed to parse `account_id`"),
```